contracts/StablecoinConverter.sol

```diff
@@ -165,9 +169,18 @@ contract StablecoinConverter is EpochTokenLocker {
                 volumes[i]
             );
         }
+        checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, prices, tokenIdsForPrice);
     }
 
+    function checkTokenConservation(
+        int[] memory tokenConservation
+    ) internal view {
+        for (uint i = 0; i < tokenConservation.length; i++) {
+            require(tokenConservation[i] == 0, "Token conservation does not hold");
```

Comment:
This is likely not going to work in practice. I think we need an epsilon inequality here. Maybe @twalth3r can give us some idea of how much we need.